testsupport/wait/host.go

```diff
@@ -2166,3 +2166,40 @@ func EncodeUserIdentifier(subject string) string {
 
 	return encoded
 }
+
+// CreateSpaceAndSpaceBinding creates a space and spacebindig and waits until both are present.
+// We are creating both of them (Space and SpaceBinding) at the same time , with polling logic, so that we mitigate the issue with spacecleanup_controller deleting the Space before we create it's SpaceBinding.
+func (a *HostAwaitility) CreateSpaceAndSpaceBinding(t *testing.T, mur *toolchainv1alpha1.MasterUserRecord, space *toolchainv1alpha1.Space, spaceRole string) (*toolchainv1alpha1.Space, *toolchainv1alpha1.SpaceBinding, error) {
+	var spaceBinding *toolchainv1alpha1.SpaceBinding
+	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
+		// create the space
+		if err := a.CreateWithCleanup(t, space); err != nil {
+			if errors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}
+		// create spacebinding request immediately after ...
+		spaceBinding = spacebinding.NewSpaceBinding(mur, space, spaceRole)
+		if err := a.CreateWithCleanup(t, spaceBinding); err != nil {
+			return false, err
+		}
+		// let's see if space was provisioned as expected
+		space, err = a.WaitForSpace(t, space.Name, UntilSpaceHasAnyTargetClusterSet(), UntilSpaceHasAnyTierNameSet())
+		if err != nil {
+			return false, err
+		}
+		// let's see if spacebinding was provisioned as expected
+		spaceBinding, err = a.WaitForSpaceBinding(t, mur.Name, space.Name,
+			UntilSpaceBindingHasMurName(mur.Name),
+			UntilSpaceBindingHasSpaceName(space.Name),
+			UntilSpaceBindingHasSpaceRole(spaceRole),
+		)
```

Comment:
Let's only check the existence in this poll loop - because this is what matters now (if it was successfully created or unexpectedly deleted). And move the verification of the content out of this function.

Reply:
Yes, let's check:
* it's present and not terminating
* if it's terminating, then wait until it is gone and do the loop again.

Reply:
I did that plus the other fixes (cleanup function for spaces and fixed the `r.WithCleanup` param in setup_runner.go)

Please check:  https://github.com/codeready-toolchain/toolchain-e2e/pull/689/commits/5654e56aad5992b6caa3f7de34f1cb9f76911220

I've still faced one issue locally, this time with the `user_workloads_test` 😕 :
```
r_workloads_test.go:78:
        	Error Trace:	/Users/fmuntean/go/src/github.com/codeready-toolchain/toolchain-e2e/test/e2e/user_workloads_test.go:78
        	Error:      	Received unexpected error:
        	            	notifications.toolchain.dev.openshift.com "test-idler-dev-idled" not found
        	Test:       	TestUserWorkloads/TestIdlerAndPriorityClass
```
but let's see if this happens also in openshift-ci job. Meantime I'm running more tests locally to check the overall stability.